app/lib/katello/validators/hostgroup_kickstart_repository_validator.rb

```diff
@@ -19,12 +19,25 @@ def validate(hostgroup)
               elsif hostgroup.architecture.blank?
                 _("Please select an architecture before assigning a kickstart repository")
               elsif hostgroup.operatingsystem.kickstart_repos(hostgroup).none? { |repo| repo[:id] == hostgroup.kickstart_repository_id }
-                _("The selected kickstart repository is not part of the assigned content view, lifecycle environment,
-                  content source, operating system, and architecture")
+                if (equivalent = find_equivalent_kickstart(hostgroup))
+                  hostgroup.kickstart_repository_id = equivalent[:id]
```

Comment:
I have mixed feelings about modifying attributes inside a validation.  I think it might be more obvious if this happened `before_validation`, what do you think?

Reply:
Yeah for sure!